dotnet-setversion/Program.cs

```diff
@@ -9,24 +9,54 @@ class Program
     {
         static int Main(string[] args)
         {
-            if (args.Length < 1)
+            string csprojFile;
+            string versionString;
+
+            bool IsHelpFlag(string arg) => arg == "-h" || arg == "--help";
+            bool IsVstsFlag(string arg) => arg == "--vsts";
+            void PrintUsage() => Console.WriteLine("Usage: (dotnet setversion <version>) || (dotnet setversion <csproj-path> <version> --vsts)");
+
+            if (args.Any(IsHelpFlag))
             {
-                Console.WriteLine("Missing version string");
-                return 1;
+                PrintUsage();
+                return 0;
             }
 
-            var versionString = args[0];
-            var currentDirectory = Directory.GetCurrentDirectory();
-            var csprojFile = Directory.EnumerateFileSystemEntries(currentDirectory, "*.csproj", SearchOption.TopDirectoryOnly).First();
-            var document = XDocument.Load(csprojFile);
+            switch (args.Length)
+            {
+                case 3 when IsVstsFlag(args[2]):
+                    csprojFile = args[0];
+                    versionString = args[1];
+                    break;
+
+                case 1 when !IsVstsFlag(args[0]):
+                    var currentDirectory = Directory.GetCurrentDirectory();
+                    csprojFile = Directory
+                        .EnumerateFileSystemEntries(currentDirectory, "*.csproj", SearchOption.TopDirectoryOnly)
+                        .FirstOrDefault();
```

Comment:
This isn't the behaviour I would expect. If there were multiple csproj in the directory, I would except setversion to fail when invoked without explicitly specifying one file. Also you'll most likely get a `NullReferenceException` if setversion gets invoked in a directory with no csproj file at all.

I would use `SingleOrDefault()` and then check `csprojFile` for `null`. If it's `null`, there are multiple csproj files so the execution should be aborted with an appropriate error message.
However, this change should be considered breaking.

Reply:
That seems fair, I'd raise this as a new issue though. This branch is pretty far behind master at the moment so I'll likely scrap the existing work I've done towards this feature and start fresh from what I've got on master right now.

Reply:
Fair enough, will do.